joomla/joomla-cms
# Update accessibility plugin to version 4.5.3

The developer of the accessibility plugin (Ran Buchnik) actualized the code using typescript. In parallel we had a GSoC student working on improvements for the plugin. In the last weeks we were able to put almost all improvements together and the result should now be integrated in Joomla.

### Summary of Changes

- Fix: Accessibility plugin is not accessible => Made a11y accessible with keyboard.
- Fix: Plugin System - Additional Accessibility Features is not correctly detected in the page landmarks by Skip To plugin
- Made text-to-speech-language and speech-to-text language change to the currently active language of the site. Default is en-US.     If the language is not supported then tts will not be displayed.
- Fixed the italic icons and corrected the emoji for underline link
- Increase / decrease line height
- Pause animations: Added the pause animations button for disabling autoplaying gifs, videos, CSS transitions, and animation.
- Reading velocity
- New feature: change the position of the icon (bottom right, bottom left, top right, top left)

### Testing Instructions

- Install this PR and run npm ci or install the package
- Just check that the close and reset icons are not italic anymore and icon for underline links is correct.
- Check that all already existent functions of the plugin are still working.
- Check if you can reach the icon using the keyboard.
- Check the speech velocity.
- Change to a language other than en-US. Check if the voice of the text-to-speech adapts to the site language or not.
    If the tts is not displayed then can confirm from here: https://jsbin.com/wuqumukesu/edit?js,console,output
    that language is not supported by your browser/os.
- Just check you are able to increase/decrease line height. Do not forget to check for persistent session and reset button.
- Make sure that you are able to play/stop gifs, autoplaying videos, CSS transitions and animations by clicking on the "Stop animation" button.
- Check both options: Emojis and Google Material Font
- Play with the position 


### Actual result BEFORE applying this Pull Request
![image](https://user-images.githubusercontent.com/9153168/200167694-8fa50480-12b1-42d7-8e77-dadb00dfd793.png)

![image](https://user-images.githubusercontent.com/9153168/200168018-e27b4f62-1399-4f96-8613-2f4c07e31647.png)


### Expected result AFTER applying this Pull Request
![image](https://user-images.githubusercontent.com/9153168/200167673-06543ac4-a8b8-4e46-a688-5b19d02780d4.png)


![image](https://user-images.githubusercontent.com/9153168/200167995-6cb307f9-3b39-455e-8427-09c3a6cbb20c.png)


### Link to documentations
Please select:
